src/GraphQL/Types/Scalars/ScalarGraphType.cs

```diff
@@ -93,5 +94,56 @@ public virtual bool CanParseValue(object value)
                 return false;
             }
         }
+
+        /// <summary>
+        /// Checks that the provided value is a valid default value.
+        /// This method should not throw an exception.
+        /// </summary>
+        public virtual bool IsValidDefault(object value)
+        {
+            try
+            {
+                return ToAst(value) != null;
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Converts a value to an AST representation. This is necessary for introspection queries
+        /// to return the default value for fields of this scalar type. This method may throw an exception
+        /// or return null for a failed conversion.
+        /// </summary>
+        public virtual IValue ToAst(object value)
```

Comment:
Also I'm not sure about `ToAst` vs `ToAST`. AST is a stable wording for **A**bstract **S**ystax **T**ree. All comments use `AST` notation.

Reply:
lol - I had it the other way before.  Based on the capitalization conventions, acronyms over two letters in length should be pascal case, such as `Asp` and `Mvc` and `Html`.

https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions

I'm fine either way, and we should keep it consistent with the rest of the library's naming convention.  You think AST is more appropriate?

Reply:
Search `Ast` vs `AST`:

```
Ast: Matching lines: 185    Matching files: 45    Total files searched: 785
AST: Matching lines: 761    Matching files: 231    Total files searched: 785
```

Reply:
AST wins 🥇

Reply:
Note that `AstFromValue` (which we can rename) had not previously been named `ASTFromValue`

Reply:
Renamed to ToAST

Reply:
I know. `AST` looks better in the end. In the beginning we can use `Ast`.